twilight-model/src/guild/afk_timeout.rs

```diff
@@ -0,0 +1,134 @@
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+/// Duration of a user being AFK before being timed out from a voice channel.
+///
+/// This value is configured [for guilds][`Guild::afk_timeout`].
+///
+/// # Examples
+///
+/// ```
+/// use twilight_model::guild::AfkTimeout;
+///
+/// assert_eq!(300, AfkTimeout::FIVE_MINUTES);
+/// ```
+///
+/// [`Guild::afk_timeout`]: super::Guild::afk_timeout
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[non_exhaustive]
+pub struct AfkTimeout(u16);
+
+impl AfkTimeout {
+    /// AFK timeout of one minute.
+    pub const ONE_MINUTE: Self = Self(60);
+
+    /// AFK timeout of five minutes.
+    pub const FIVE_MINUTES: Self = Self(300);
+
+    /// AFK timeout of fifteen minutes.
+    pub const FIFTEEN_MINUTES: Self = Self(900);
+
+    /// AFK timeout of thirty minutes.
+    pub const THIRTY_MINUTES: Self = Self(1800);
+
+    /// AFK timeout of one hour.
+    pub const ONE_HOUR: Self = Self(3600);
+
+    /// Retrieve the duration of the AFK timeout in seconds.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use twilight_model::guild::AfkTimeout;
+    ///
+    /// assert_eq!(60, AfkTimeout::ONE_MINUTE.get());
+    /// ```
+    pub const fn get(self) -> u16 {
+        self.0
+    }
+}
+
+impl From<u16> for AfkTimeout {
+    fn from(value: u16) -> Self {
+        Self(value)
+    }
+}
```

Comment:
Just highlighting that this allows any value

Reply:
I think this is fine since it's non exaustive and we have no guarantee discord won't add any extra values later